### PR TITLE
docs(skill): enhance grill-me with domain-awareness and doc updates

### PR DESCRIPTION
**Pull request by macalbert**
# Pull Request

## What does this PR do?

Enhances the existing `grill-me` skill by merging domain-awareness and inline documentation behaviors from Matt Pocock's [`grill-with-docs`](https://github.com/mattpocock/skills/tree/main/skills/engineering/grill-with-docs) skill.

Added capabilities:
- Challenge user terminology against existing `CONTEXT.md` glossary
- Sharpen fuzzy/overloaded language into precise canonical terms
- Stress-test domain relationships with concrete edge-case scenarios
- Cross-reference user statements against actual code
- Inline `CONTEXT.md` updates as terms are resolved
- Offer ADRs sparingly (only when hard to reverse, surprising, and a real trade-off)

New supporting files:
- `CONTEXT-FORMAT.md` — Template and rules for maintaining a domain glossary
- `ADR-FORMAT.md` — Minimal ADR template with qualification criteria

## Related issues

N/A

## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Refactor
- [x] Documentation
- [ ] Other

## Checklist

- [ ] Tests added/updated (if needed)
- [x] Docs updated (if needed)
- [x] Lint/format pass

## Notes for reviewer

Based on [grill-with-docs](https://github.com/mattpocock/skills/tree/main/skills/engineering/grill-with-docs) by [@mattpocock](https://github.com/mattpocock). Attribution included in SKILL.md footer and via `Co-authored-by` commit trailer.

### CODE REVIEW

File: .github/skills/grill-me/CONTEXT-FORMAT.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{A one or two sentence description of the term}
+_Avoid_: Purchase, transaction
+
+**Invoice**:
+A request for payment sent to a customer after delivery.
+_Avoid_: Bill, payment request
+
+**Customer**:
+A person or organization that places orders.
+_Avoid_: Client, buyer, account
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/CONTEXT-FORMAT.md]: **Resolved.** Updated `CONTEXT-FORMAT.md` to use Markdown table format (`Term | Definition | Avoid`) matching the existing `CONTEXT.md` structure.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)

File: .github/skills/grill-me/SKILL.md [hunk 2]
```diff
@@ -19,3 +27,95 @@ dependencies between decisions one by one.
...
+
+Most repos have a single context:
+
+```
+/
+├── CONTEXT.md
+├── docs/
+│   └── adr/
+│       ├── 0001-event-sourced-orders.md
+│       └── 0002-postgres-for-write-model.md
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/SKILL.md:45]: **Resolved.** Standardized all ADR path references to `docs/adr/` across SKILL.md, ADR-FORMAT.md, and CONTEXT-FORMAT.md.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)

File: .github/skills/grill-me/SKILL.md [hunk 3]
```diff
@@ -19,3 +27,95 @@ dependencies between decisions one by one.
...
+│   │   ├── CONTEXT.md
+│   │   └── docs/adr/                 ← context-specific decisions
+│   └── billing/
+│       ├── CONTEXT.md
+│       └── docs/adr/
+```
+
+Create files lazily — only when you have something to write. If no `CONTEXT.md`
+exists, create one when the first term is resolved. If no `docs/adr/` exists,
+create it when the first ADR is needed.
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/SKILL.md:68]: **Resolved.** Same fix — all references now point to `docs/adr/`.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)

File: .github/skills/grill-me/CONTEXT-FORMAT.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/CONTEXT-FORMAT.md:17]: **Resolved.** Same fix — CONTEXT-FORMAT now uses table format matching repo conventions.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)

File: .github/skills/grill-me/ADR-FORMAT.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+Create the `docs/adr/` directory lazily — only when the first ADR is needed.
+
+## Template
+
+```md
+# {Short title of the decision}
+
+{1-3 sentences: what's the context, what did we decide, and why.}
+```
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/ADR-FORMAT.md:14]: **Resolved.** ADR template now aligned. Path standardized to `docs/adr/`.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)

File: .github/skills/grill-me/ADR-FORMAT.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+- **Status** frontmatter (`proposed | accepted | deprecated | superseded by
+  ADR-NNNN`) — useful when decisions are revisited
+- **Considered Options** — only when the rejected alternatives are worth
+  remembering
+- **Consequences** — only when non-obvious downstream effects need to be called
+  out
+
+## Numbering
+
+Scan `docs/adr/` for the highest existing number and increment by one.
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/ADR-FORMAT.md:32]: **Resolved.** Numbering scan path updated to `docs/adr/`.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)

File: .github/skills/grill-me/ADR-FORMAT.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
+# ADR Format
+
+ADRs live in `docs/adr/` (or `docs/architecture/adr/`) and use sequential
+numbering: `0001-slug.md`, `0002-slug.md`, etc.
+
+Create the `docs/adr/` directory lazily — only when the first ADR is needed.
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/ADR-FORMAT.md:6]: **Resolved.** Removed `docs/architecture/adr/` fallback — now hard-codes `docs/adr/` only.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)

File: .github/skills/grill-me/ADR-FORMAT.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+That's it. An ADR can be a single paragraph. The value is in recording *that* a
+decision was made and *why* — not in filling out sections.
+
+## Optional Sections
+
+Only include these when they add genuine value. Most ADRs won't need them.
+
+- **Status** frontmatter (`proposed | accepted | deprecated | superseded by
+  ADR-NNNN`) — useful when decisions are revisited
```
[the remark this answers is not in the document]
Comment (macalbert) [reply .github/skills/grill-me/ADR-FORMAT.md:24]: **Resolved.** Status section description updated to match repo conventions.

**Commit:** [`8ba450f`](https://github.com/macalbert/envilder/commit/8ba450f)
